src/bio.c

```diff
@@ -220,7 +220,13 @@ void *bioProcessBackgroundJobs(void *arg) {
         if (type == BIO_CLOSE_FILE) {
             close(job->fd);
         } else if (type == BIO_AOF_FSYNC) {
-            redis_fsync(job->fd);
+            if (redis_fsync(job->fd) == -1) {
+                atomicSet(server.aof_bio_fsync_status,C_ERR);
+                serverLog(LL_WARNING,
+                    "Fail to fsync the AOF file: %s",strerror(errno));
```

Comment:
are we sure we wanna log every time fsync fails (i fear of flooding the log file on such errors)? maybe just on state change?
although i suppose soon enough we'll stop accepting writes, right? and thus stop the fsync attempts and logging...

Reply:
Yes, in this case, we want to tell users fsync errors and handle ASAP, i agree it is a good way to only log this error on state changed.